gobblin-data-management/src/test/java/gobblin/data/management/retention/TimeBasedRetentionPolicyTest.java

```diff
@@ -54,4 +61,48 @@ public void test() {
 
     DateTimeUtils.setCurrentMillisSystem();
   }
+
+  @Test(dependsOnMethods = { "testDefaultRetention" })
+  public void testISORetentionDuration() throws Exception {
+
+    DateTimeUtils.setCurrentMillisFixed(new DateTime(2016, 2, 11, 10, 0, 0, 0).getMillis());
+
+    // 20 Days
+    verify("P20D", new TimestampedDatasetVersion(new DateTime(2016, 2, 10, 10, 0, 0, 0), new Path("test1")),
+        new TimestampedDatasetVersion(new DateTime(2016, 1, 5, 10, 0, 0, 0), new Path("test2")));
+
+    // 2 Days
+    verify("P2M", new TimestampedDatasetVersion(new DateTime(2016, 2, 10, 10, 0, 0, 0), new Path("test1")),
+        new TimestampedDatasetVersion(new DateTime(2015, 12, 5, 10, 0, 0, 0), new Path("test2")));
+
+    // 2 Years
+    verify("P2Y", new TimestampedDatasetVersion(new DateTime(2016, 2, 10, 10, 0, 0, 0), new Path("test1")),
+        new TimestampedDatasetVersion(new DateTime(2014, 1, 5, 10, 0, 0, 0), new Path("test2")));
+
+    // 20 Hours
+    verify("PT20H", new TimestampedDatasetVersion(new DateTime(2016, 2, 11, 8, 0, 0, 0), new Path("test1")),
+        new TimestampedDatasetVersion(new DateTime(2016, 2, 10, 11, 0, 0, 0), new Path("test2")));
+
+    // 24 Hours
+    verify("PT20H", new TimestampedDatasetVersion(new DateTime(2016, 2, 11, 8, 0, 0, 0), new Path("test1")),
+        new TimestampedDatasetVersion(new DateTime(2016, 2, 10, 9, 59, 0, 0), new Path("test2")));
+
+    DateTimeUtils.setCurrentMillisSystem();
+  }
+
+  private void verify(String duration, TimestampedDatasetVersion... versions) {
+    Config config = ConfigFactory.parseMap(ImmutableMap.<String, String>of(TimeBasedRetentionPolicy.RETENTION_TIMEBASED_DURATION_KEY, duration));
+    TimeBasedRetentionPolicy policy = new TimeBasedRetentionPolicy(config);
+
+    List<TimestampedDatasetVersion> versionsList = Lists.newArrayList(versions);
+
+    TimestampedDatasetVersion datasetVersion1 = versionsList.get(1);
+
+    List<TimestampedDatasetVersion> deletableVersions = Lists.newArrayList(policy.listDeletableVersions(versionsList));
+
+    Assert.assertEquals(deletableVersions.size(), 1);
+
+    Assert.assertEquals(deletableVersions.get(0).getDateTime(), datasetVersion1.getDateTime());
```

Comment:
In sync with default retention testing, prefer to add both "should delete" and "should keep" TimestampedDatasetVersion